lib/ansible/modules/cloud/vmware/vmware_guest_sendkey.py

```diff
@@ -0,0 +1,357 @@
+#!/usr/bin/python
+#  -*- coding: utf-8 -*-
+#  Copyright: (c) 2018, Ansible Project
+#  Copyright: (c) 2018, Diane Wang <dianew@vmware.com>
+#  GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+ANSIBLE_METADATA = {
+    'metadata_version': '1.1',
+    'status': ['preview'],
+    'supported_by': 'community'
+}
+
+DOCUMENTATION = '''
+---
+module: vmware_guest_sendkey
+short_description: Send USB HID codes to the Virtual Machine's keyboard.
+description:
+    - This module is used to send keystrokes to given virtual machine.
+    - All parameters and VMware object names are case sensitive.
+version_added: '2.9'
+author:
+    - Diane Wang (@Tomorrow9) <dianew@vmware.com>
+notes:
+    - Tested on vSphere 6.5 and 6.7
+requirements:
+    - "python >= 2.6"
+    - PyVmomi
+options:
+   name:
+     description:
+     - Name of the virtual machine.
+     - This is a required parameter, if parameter C(uuid) is not supplied.
+   uuid:
+     description:
+     - UUID of the instance to gather facts if known, this is VMware's unique identifier.
+     - This is a required parameter, if parameter C(name) is not supplied.
+   folder:
+     description:
+     - Destination folder, absolute or relative path to find an existing guest.
+     - This is a required parameter, only if multiple VMs are found with same name.
+     - The folder should include the datacenter. ESXi server's datacenter is ha-datacenter.
+     - 'Examples:'
+     - '   folder: /ha-datacenter/vm'
+     - '   folder: ha-datacenter/vm'
+     - '   folder: /datacenter1/vm'
+     - '   folder: datacenter1/vm'
+     - '   folder: /datacenter1/vm/folder1'
+     - '   folder: datacenter1/vm/folder1'
+     - '   folder: /folder1/datacenter1/vm'
+     - '   folder: folder1/datacenter1/vm'
+     - '   folder: /folder1/datacenter1/vm/folder2'
+   cluster:
+     description:
+     - The name of cluster where the virtual machine will run.
+     - This is a required parameter, if C(esxi_hostname) is not set.
+     - C(esxi_hostname) and C(cluster) are mutually exclusive parameters.
+   esxi_hostname:
+     description:
+     - The ESXi hostname where the virtual machine will run.
+     - This is a required parameter, if C(cluster) is not set.
+     - C(esxi_hostname) and C(cluster) are mutually exclusive parameters.
+   datacenter:
+     default: ha-datacenter
```

Comment:
I tend to avoid default for datacenter since it causes unwanted result. In my opinion it is OK to raise an error message to user rather than setting a default value that will give unwanted results.

Reply:
Indeed, I'll modify this. Thank you for reviewing.